test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,338 @@
+    cluster1_.cluster_info_->stats().upstream_rq_per_try_timeout_.add(2);
+    cluster1_.cluster_info_->stats().upstream_rq_pending_overflow_.add(8);
+
+    // Cluster 2
+    ON_CALL(cluster2.error_5xx_counter_, value()).WillByDefault(Return((i + 1) * 1));
+    ON_CALL(cluster2.retry_5xx_counter_, value()).WillByDefault(Return((i + 1) * 2));
+    ON_CALL(cluster2.error_4xx_counter_, value()).WillByDefault(Return((i + 1) * 3));
+    ON_CALL(cluster2.retry_4xx_counter_, value()).WillByDefault(Return((i + 1) * 4));
+    ON_CALL(cluster2.success_counter_, value()).WillByDefault(Return((i + 1) * 3));
+    cluster2.cluster_info_->stats().upstream_rq_timeout_.add(3);
+    cluster2.cluster_info_->stats().upstream_rq_per_try_timeout_.add(1);
+    cluster2.cluster_info_->stats().upstream_rq_pending_overflow_.add(5);
+
+    sink_->flush(source_);
+  }
+
+  std::unordered_map<std::string, std::string> cluster_message_map =
+      buildClusterMap(TestUtility::bufferToString(buffer));
+  ASSERT_NE(cluster_message_map.find(cluster1_name_), cluster_message_map.end());
+  ASSERT_NE(cluster_message_map.find(cluster2_name), cluster_message_map.end());
+
+  std::string data_message_1 = cluster_message_map[cluster1_name_];
+  std::string data_message_2 = cluster_message_map[cluster2_name];
+
+  // Check stream format and data.
+  EXPECT_EQ(getStreamField(data_message_1, "errorCount"), "160"); // Note that on regular operation,
+  // 5xx and timeout are raised together, so timeouts are reduced from 5xx count.
+  EXPECT_EQ(getStreamField(data_message_1, "requestCount"), "340");
+  EXPECT_EQ(getStreamField(data_message_1, "rollingCountSemaphoreRejected"), "80");
+  EXPECT_EQ(getStreamField(data_message_1, "rollingCountSuccess"), "70");
+  EXPECT_EQ(getStreamField(data_message_1, "rollingCountTimeout"), "30");
+  EXPECT_EQ(getStreamField(data_message_1, "errorPercentage"), "79");
+
+  // Check stream format and data.
+  EXPECT_EQ(getStreamField(data_message_2, "errorCount"), "70"); // note that on regular operation,
+                                                                 // 5xx and timeout are raised
+                                                                 // together, so timeouts are
+                                                                 // reduced from 5xx count
+  EXPECT_EQ(getStreamField(data_message_2, "requestCount"), "190");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountSemaphoreRejected"), "50");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountSuccess"), "30");
+  EXPECT_EQ(getStreamField(data_message_2, "rollingCountTimeout"), "40");
+  EXPECT_EQ(getStreamField(data_message_2, "errorPercentage"), "84");
+
+  buffer.drain(buffer.length());
+
+  // Removing cluster.
+  cluster_map_.erase(cluster2_name);
+  // Redefining since cluster_map_ is returned by value.
+  ON_CALL(cluster_manager_, clusters()).WillByDefault(Return(cluster_map_));
+  sink_->flush(source_);
+
+  cluster_message_map.clear();
```

Comment:
you don't need this, as on the next line you kill the previous value completely.